### PR TITLE
Improve logging for HttpProxy async requests

### PR DESCRIPTION
**Pull request by matthiaso**
Copy run monitor (or create a new one) to include correlation id and further diagnostics information.

374142

### CODE REVIEW

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java [hunk 1]
```diff
@@ -147,4 +153,43 @@ protected synchronized void removeClient() {
       m_initialized = false;
     }
   }
+
+  /**
+   * Proxy the actual object using the {@link RunContext} at time of proxy construction for all method calls. All method
+   * calls on the proxy are wrapped with the specific {@link RunContext}.
+   */
+  public <T> T createAsyncInvocationHandler(Class<T> clazz, T actualObject) {
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java:162]: maybe add a comment about the purpose for this wrapping into  a proxy:  allow to retain context infos for logging?

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java [hunk 2]
```diff
@@ -147,4 +153,43 @@ protected synchronized void removeClient() {
...
+  }
+
+  public static class AsyncHttpInvocationHandler<T> implements InvocationHandler {
+
+    private final RunContext m_runContext;
+    private final T m_actualObject;
+
+    public AsyncHttpInvocationHandler(T actualObject) {
+      m_runContext = RunContexts.copyCurrent(true)
+          .withTransactionScope(TransactionScope.REQUIRES_NEW);
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java:180]: why force to require new here? avoid to re-using existing transaction of current thread? -> add comment?

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/DefaultAsyncHttpClientManager.java [hunk 3]
```diff
@@ -82,6 +84,11 @@ protected void interceptCreateClient(HttpAsyncClientBuilder builder) {
     setConnectionEvictionSettings(builder);
 
     builder.setConnectionManager(createConnectionManager());
+
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/DefaultAsyncHttpClientManager.java:87]: add javadoc explaing this (rather complicated line of code), 
e.g. "adding execution interceptor allowing access to entityProducer and asyncExecCallback e.g. for logging puposes"

File: org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/servlet/HttpProxy.java [hunk 4]
```diff
@@ -629,14 +629,17 @@ protected void produceData(ContentType contentType, OutputStream outputStream) t
    * </p>
    */
   protected AsyncEntityConsumer<Boolean> createEntityConsumer(HttpServletResponse resp) {
-    return new AbstractClassicEntityConsumer<>(getInitialBufferSize(resp), getBlockingOperationExecutor()) {
+    AsyncEntityConsumer<Boolean> entityConsumer = new AbstractClassicEntityConsumer<>(getInitialBufferSize(resp), getBlockingOperationExecutor()) {
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/servlet/HttpProxy.java:632]: in which cases is this manual wrapping necessary when using an http client manger instance which is setup with an exec-interceptor?  --> add java doc here to explan?  (and also in other methods of http proxy?=

**Review by paolobazzi — APPROVED** [review]
looks good, see comments for minor improvements

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java (hunk 1 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java:162]: Added javadoc/comment

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/DefaultAsyncHttpClientManager.java (hunk 3 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/DefaultAsyncHttpClientManager.java:87]: Added javadoc/comment

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java (hunk 2 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/async/AbstractAsyncHttpClientManager.java:180]: Added javadoc/comment

File: org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/servlet/HttpProxy.java (hunk 4 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/servlet/HttpProxy.java:632]: Added javadoc/comment

**Pull request merged**